### PR TITLE
mim-solvers: init at 0.0.5

### PR DESCRIPTION
**Pull request by nim65s**
## Description of changes

This add [mim-solvers](https://github.com/machines-in-motion/mim_solvers), an "Implementation of numerical solvers" used in the Machines in Motion Laboratory at  NYU.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by FliegendeWurst — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
Please check ofborg `aarch64-linux` build failure. Can we disable that test?

File: pkgs/by-name/mi/mim-solvers/package.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    substituteInPlace CMakeLists.txt --replace-fail \
+      "add_project_dependency(eigenpy 2.7.10 REQUIRED)" \
+      ""
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+  buildInputs = lib.optional stdenv.isDarwin llvmPackages.openmp;
```
Comment (FliegendeWurst) [inline pkgs/by-name/mi/mim-solvers/package.nix:36]:
```suggestion
  buildInputs = lib.optional stdenv.hostPlatform.isDarwin llvmPackages.openmp;
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/by-name/mi/mim-solvers/package.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    pkg-config
+  ];
+  buildInputs = lib.optional stdenv.hostPlatform.isDarwin llvmPackages.openmp;
+  propagatedBuildInputs =
+    lib.optionals pythonSupport [
+      python3Packages.crocoddyl
+      python3Packages.osqp
+      python3Packages.proxsuite
+      python3Packages.scipy
+      python3Packages.pythonImportsCheckHook
```
Comment (natsukium) [inline pkgs/by-name/mi/mim-solvers/package.nix:43]: `pythonImportsCheckHook` should go to `nativeBuildInputs`.

**Comment by natsukium** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 345494`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>mim-solvers</li>
    <li>python311Packages.mim-solvers</li>
    <li>python312Packages.mim-solvers</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>mim-solvers</li>
    <li>python311Packages.mim-solvers</li>
    <li>python312Packages.mim-solvers</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>mim-solvers</li>
    <li>python311Packages.mim-solvers</li>
    <li>python312Packages.mim-solvers</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>mim-solvers</li>
    <li>python311Packages.mim-solvers</li>
    <li>python312Packages.mim-solvers</li>
  </ul>
</details>

**Pull request merged**